src/pages/Cart.tsx

```diff
@@ -1,39 +1,71 @@
-import { useState } from 'react';
-import './PagesStyles/Cart.scss';
+import { useState, useContext, useMemo, useEffect, useCallback } from 'react';
 import { Checkout } from '../components/Cart/Checkout/Checkout';
-import ProductsInCart from './../phonesBase.json';
 import { CartList } from '../components/Cart/Cartlist/CartList';
-import { CartModal } from '../components/Cart/CartModal/Cartmodal';
-
-const threeProductsInCart = ProductsInCart.slice(0, 3);
-
-export const Cart = () => {
-  // const { cartContents } = useContext(StoreContext);
+import { CartModal } from '../components/Cart/CartModal/CartModal';
+import { Loader } from '../components/Loader';
+import { StoreContext } from '../context/StoreContext';
+import { CatalogProductData } from '../types/CatalogProductData';
+import { getProductsForCart } from '../api';
+import './PagesStyles/Cart.scss';
 
 export const Cart = () => {
+  const [products, setProducts] = useState<CatalogProductData[]>([]);
   const [showModal, setShowModal] = useState(false);
+  const { cartContents } = useContext(StoreContext);
 
   const onCheckout = () => {
     setTimeout(() => {
       setShowModal(true);
     }, 100);
   };
 
+  const phoneIds = cartContents.map(({ id }) => id).join(',');
+
+  const getCartContents = useCallback(async() => {
```

Comment:
Forgot about loading process and about Loader component

Reply:
Loader is present down the code. But its logic needs to be adjusted for sure